src/server.rs

```diff
@@ -259,26 +260,43 @@ impl<L, M> Server<L, M> where L: Log, M: StateMachine {
                 ConnectionKind::Unknown => {
                     let preamble = try!(message.get_root::<connection_preamble::Reader>());
                     match try!(preamble.get_id().which()) {
-                        connection_preamble::id::Which::Server(id) => {
-                            let peer_id = ServerId(id);
-                            scoped_debug!("received new connection from {:?}", peer_id);
+                        connection_preamble::id::Which::Server(peer) => {
+                            let peer = peer.unwrap(); // TODO(tschottdorf): nicer way?
```

Comment:
ah actually I think it's easier than that.  If you return a suitable `Err` instance it should be automatically reset by `ready` or whoever calls this method.